src/pages/billing/Project.js

```diff
@@ -475,30 +459,38 @@ const ProjectDetail = ({ project, billingAccounts, authorizeAndLoadAccounts }) =
       footer: 'Warning: Adding any user to this project will mean they can incur costs to the billing associated with this project.',
       addFunction: Ajax().Billing.project(billingProject.projectName).addUser,
       onDismiss: () => setAddingUser(false),
-      onSuccess: refresh
+      onSuccess: () => {
+        setAddingUser(false)
+        updateBillingProjectUsers()
+      }
     }),
     editingUser && h(EditUserModal, {
       adminLabel: billingRoles.owner,
       userLabel: billingRoles.user,
       user: editingUser,
       saveFunction: Ajax().Billing.project(billingProject.projectName).changeUserRoles,
       onDismiss: () => setEditingUser(false),
-      onSuccess: refresh
+      onSuccess: () => {
+        setEditingUser(false)
+        updateBillingProject()
+        updateBillingProjectUsers()
+      }
     }),
     !!deletingUser && h(DeleteUserModal, {
       userEmail: deletingUser.email,
       onDismiss: () => setDeletingUser(false),
-      onSubmit: _.flow(
-        withErrorReporting('Error removing member from billing project'),
-        Utils.withBusyState(setUpdating)
-      )(async () => {
+      onSubmit: async () => {
+        await _.flow(
```

Comment:
Might this be one of those places where it's useful to name this function, since it has all these wrappers around it, and then invoke it by name? Thinking purely of ease of readability.

Reply:
I completely agree. Thanks.